pkg/manifest/manifest.go

```diff
@@ -132,15 +169,17 @@ func (m *Manifest) Include(excludeIdentifier *string, includeTechPreview *bool,
 		}
 	}
 
-	if includeTechPreview != nil {
-		featureGateAnnotation := "release.openshift.io/feature-gate"
-		featureGateAnnotationValue, featureGateAnnotationExists := annotations[featureGateAnnotation]
-		if featureGateAnnotationValue == string(configv1.TechPreviewNoUpgrade) && !(*includeTechPreview) {
-			return fmt.Errorf("tech-preview excluded, and %s=%s", featureGateAnnotation, featureGateAnnotationValue)
+	if requiredFeatureSet != nil {
+		requiredAnnotationValue := *requiredFeatureSet
+		if len(*requiredFeatureSet) == 0 {
+			requiredAnnotationValue = "Default" // "" in the FeatureSet API is "Default" in the annotation value
+		}
+		manifestFeatureSets, manifestSpecifiesFeatureSets, err := getFeatureSets(annotations)
+		if err != nil {
+			return err
 		}
-		// never include the manifest if the feature-gate annotation is outside of allowed values (only TechPreviewNoUpgrade is currently allowed)
-		if featureGateAnnotationExists && featureGateAnnotationValue != string(configv1.TechPreviewNoUpgrade) {
-			return fmt.Errorf("unrecognized value %s=%s", featureGateAnnotation, featureGateAnnotationValue)
+		if manifestSpecifiesFeatureSets && !manifestFeatureSets.Has(requiredAnnotationValue) {
+			return fmt.Errorf("%q is required, and %s=%s", requiredAnnotationValue, featureSetAnnotation, strings.Join(manifestFeatureSets.List(), ","))
```

Comment:
nit: all of this could be pushed down into a `checkFeatureSets`, like:

```go
if  requiredFeatureSet != nil {
  err := checkFeatureSets(annotations, requiredFeatureSet)
  if err != nil {
    return err
  }
}
```

which is the pattern we use today for `checkResourceEnablement`.  To implement, you could shift some of this wrapping stuff over into your existing `getFeatureSets` and then rename and shuffle the function signature.